### DISCUSSION TITLE
rabbitmq shovel default exchange

### DISCUSSION THREAD

**Discussion by lanzhiwang** [rabbitmq/rabbitmq-server#10903] [Other]
I have a special scenario. There are some queues in my rabbitmq cluster. These queues are not explicitly bound to specific exchanges. The producers directly produce messages to the queues.
That is to say, the feature of (AMQP default) default exchange is used. It is bound to all queues by default. Routes key is the queue name.

I now want to back up the messages in these queues to another rabbitmq cluster through shovel. In shovel, I need to set source_exchange to (AMQP default), as shown in the figure below

<img width="679" alt="image" src="https://github.com/rabbitmq/rabbitmq-server/assets/7719138/d5ae784e-3fd8-43c4-8bbc-eb1da4a820a0">

<img width="971" alt="image" src="https://github.com/rabbitmq/rabbitmq-server/assets/7719138/6261da85-50ed-40c5-a1f3-ee79d06ddbf5">

But (AMQP default) is not a correct exchange name, what should I set the (AMQP default) name, or can shovel meet my requirements?

**Comment by michaelklishin** [comment]
The protocol uses an empty string to identify that exchange. `amq.default` is used in a few more contexts.

You can still use individual queues as sources for N shovels. Or a different direct exchange with explicit bindings for your queues.

**Reply by gernotkogler** [reply]
Leaving the exchange names blank does not work, exchange names are mandatory when using the UI shown above.

**Reply by michaelklishin** [reply]
Without adopting the `amq.default` alias in more places, I can only suggest using the queue as the destination or an explicitly declared exchange, direct or topic.

It's too late to try to introduce `amq.default` for 4.1.0, and then we will see.

**Reply by gernotkogler** [reply]
That's how I solved it for now. Direct queue to queue shoveling. Thanks for looking into it.
